ops/services/app_service/metabase-readonly/main.tf

```diff
@@ -0,0 +1,135 @@
+# Initializing a new readonly user and granting them the proper permissions has to be done via 
```

Comment:
It looks like, aside from the `locals` and the changed `app_settings`, this is identical to `ops/services/app_service/metabase/main.tf`. I think we can avoid creating this file/directory entirely by pulling the `app_settings` out into `locals`. Then, add a `variable` that accepts a

As an example:

`ops/services/app_service/metabase/main.tf`
```
locals {
  app_settings_defaults = {
    "var1" = 1
    "var2" = 2
  }
}

variable "app_settings_overrides" {
  description = "..."
  default = {}
}

resource "azurem_app_service" "metabase" {
  ...
  app_settings = merge(local.app_settings_defaults, var.app_settings_overrides)
}
```

Then, in `ops/$ENV/metabase.tf`:

```
module "metabase" {
...
}
```

In this case, `app_settings` would be what you expected - `{"var1" = 1, "var2" = 2}`. Since we didn't override any of the defaults in the module, the `merge` function merged an empty map and they were passed to the Metabase resource unchanged.

But! Now, we want to make a read-only Metabase in addition, so we add a new section to `ops/$ENV/metabase.tf`:

```
module "metabase_readonly" {
  app_settings = {
    "var1" = 2
  }
}
```

And for this new Metabase resource, you'd end up with `{"var1" = 2, "var2" = 2}`. Well, unless we don't want to do this for every environment, in which case we'd move it to the relevant `ops/$ENV` directories (like you did for `test` in this PR).

Reply:
Thanks! This makes a lot of sense. I've tried to implement these changes, please take a look and let me know if it's what you had in mind.